From bugs+caf_=bugs=bugs.plt-scheme.org@plt-scheme.org Sat Apr  6 14:47:38 2013
Received: from mail-ob0-f179.google.com (mail-ob0-f179.google.com [209.85.214.179])
	by champlain.ccs.neu.edu (8.14.4/8.14.4) with ESMTP id r36IlZuK007249
	for <bugs@bugs.plt-scheme.org>; Sat, 6 Apr 2013 14:47:35 -0400
Message-Id: <201304061847.r36IlXIE007243@champlain.ccs.neu.edu>
Date: Sat, 6 Apr 2013 14:47:33 -0400
From: eric.n.dobson@gmail.com
To: bugs@racket-lang.org
Subject: Function contracted with (case->) gives bad error message

>Number:         13665
>Category:       all
>Synopsis:       Function contracted with (case->) gives bad error message
>Class:          sw-bug
>Responsible:    endobson
>Severity:       serious
>Priority:       medium
>State:          closed
>Confidential:   no
>Arrival-Date:   Sat Apr 06 14:48:01 -0400 2013
>Closed-Date:    Thu Apr 25 11:44:16 -0400 2013
>Last-Modified:  Thu Apr 25 11:44:16 -0400 2013
>Originator:     Eric Dobson
>Organization:
plt
>Submitter-Id:   unknown
>Release:        5.3.2.3
>Environment:
MacIntel / Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_2) AppleWebKit/537.28 (KHTML, like Gecko) Chrome/26.0.1397.2 Safari/537.28
>Description:
The following program gives the error message:

#<case-lambda-procedure>: arity mismatch;
 the expected number of arguments does not match the given number
  given: 1
  arguments...:
   'x
  context...:
   /Users/endobson/tmp/tmp2.rkt: [running body]

It does not say what the number of expected arguments is, and it doesn't have the same name as the function.
>How-To-Repeat:
#lang racket

(define/contract (f x)
  (case->)
  x)

(f 'x)
>Fix:
>Audit-Trail:
From: Robby Findler <robby@eecs.northwestern.edu>
To: Eric Dobson <eric.n.dobson@gmail.com>, bugs <bugs@racket-lang.org>
Cc: nobody <nobody@racket-lang.org>,
        bug-notification <bug-notification@racket-lang.org>
Subject: Re: [racket-bug] all/13665: Function contracted with (case->) gives
 bad error message
Date: Sat, 6 Apr 2013 14:09:19 -0500

 --001a11c20cb0d25c8b04d9b5f0bd
 Content-Type: text/plain; charset=UTF-8
 
 I'll fix the name for the procedure, tho, so that will be better, but the
 error message is not coming from the contract system, but from the
 racket/base implementation of #%app so any "expected" change would have to
 come from there.
 
 Robby
 
 
 
 On Sat, Apr 6, 2013 at 1:48 PM, <eric.n.dobson@gmail.com> wrote:
 
 > A new problem report is waiting at
 >   http://bugs.racket-lang.org/query/?cmd=view&pr=13665
 >
 > Reported by Eric Dobson for release: 5.3.2.3
 >
 > *** Description:
 > The following program gives the error message:
 >
 > #<case-lambda-procedure>: arity mismatch;
 >  the expected number of arguments does not match the given number
 >   given: 1
 >   arguments...:
 >    'x
 >   context...:
 >    /Users/endobson/tmp/tmp2.rkt: [running body]
 >
 > It does not say what the number of expected arguments is, and it doesn't
 > have the same name as the function.
 >
 > *** How to repeat:
 > #lang racket
 >
 > (define/contract (f x)
 >   (case->)
 >   x)
 >
 > (f 'x)
 >
 > *** Environment:
 > MacIntel / Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_2)
 > AppleWebKit/537.28 (KHTML, like Gecko) Chrome/26.0.1397.2 Safari/537.28
 >
 >
 
 --001a11c20cb0d25c8b04d9b5f0bd
 Content-Type: text/html; charset=UTF-8
 Content-Transfer-Encoding: quoted-printable
 
 <div dir=3D"ltr">I&#39;ll fix the name for the procedure, tho, so that will=
  be better, but the error message is not coming from the contract system, b=
 ut from the racket/base implementation of #%app so any &quot;expected&quot;=
  change would have to come from there.<div>
 <br></div><div>Robby<br><div><div><br></div></div></div></div><div class=3D=
 "gmail_extra"><br><br><div class=3D"gmail_quote">On Sat, Apr 6, 2013 at 1:4=
 8 PM,  <span dir=3D"ltr">&lt;<a href=3D"mailto:eric.n.dobson@gmail.com" tar=
 get=3D"_blank">eric.n.dobson@gmail.com</a>&gt;</span> wrote:<br>
 <blockquote class=3D"gmail_quote" style=3D"margin:0 0 0 .8ex;border-left:1p=
 x #ccc solid;padding-left:1ex">A new problem report is waiting at<br>
 =C2=A0 <a href=3D"http://bugs.racket-lang.org/query/?cmd=3Dview&amp;pr=3D13=
 665" target=3D"_blank">http://bugs.racket-lang.org/query/?cmd=3Dview&amp;pr=
 =3D13665</a><br>
 <br>
 Reported by Eric Dobson for release: 5.3.2.3<br>
 <br>
 *** Description:<br>
 The following program gives the error message:<br>
 <br>
 #&lt;case-lambda-procedure&gt;: arity mismatch;<br>
 =C2=A0the expected number of arguments does not match the given number<br>
 =C2=A0 given: 1<br>
 =C2=A0 arguments...:<br>
 =C2=A0 =C2=A0&#39;x<br>
 =C2=A0 context...:<br>
 =C2=A0 =C2=A0/Users/endobson/tmp/tmp2.rkt: [running body]<br>
 <br>
 It does not say what the number of expected arguments is, and it doesn&#39;=
 t have the same name as the function.<br>
 <br>
 *** How to repeat:<br>
 #lang racket<br>
 <br>
 (define/contract (f x)<br>
 =C2=A0 (case-&gt;)<br>
 =C2=A0 x)<br>
 <br>
 (f &#39;x)<br>
 <br>
 *** Environment:<br>
 MacIntel / Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_2) AppleWebKit/537.2=
 8 (KHTML, like Gecko) Chrome/26.0.1397.2 Safari/537.28<br>
 <br>
 </blockquote></div><br></div>
 
 --001a11c20cb0d25c8b04d9b5f0bd--
From: Robby Findler <robby@eecs.northwestern.edu>
To: Eric Dobson <eric.n.dobson@gmail.com>, bugs <bugs@racket-lang.org>
Cc: nobody <nobody@racket-lang.org>,
        bug-notification <bug-notification@racket-lang.org>
Subject: Re: [racket-bug] all/13665: Function contracted with (case->) gives
 bad error message
Date: Sat, 6 Apr 2013 14:24:43 -0500

 --089e0111be00e1a90604d9b627ef
 Content-Type: text/plain; charset=UTF-8
 
 Oh-- the actually expression that triggers the message for the example
 given in the PR is:
 
 > (apply (case-lambda) (list 'x))
 #<case-lambda-procedure>: arity mismatch;
  the expected number of arguments does not match the given number
   given: 1
   arguments...:
    'x
   context...:
    /Users/robby/git/plt/collects/racket/private/misc.rkt:87:7
 
 
 Robby
 
 --089e0111be00e1a90604d9b627ef
 Content-Type: text/html; charset=UTF-8
 Content-Transfer-Encoding: quoted-printable
 
 <div dir=3D"ltr">Oh-- the actually expression that triggers the message for=
  the example given in the PR is:<div><br></div><div><div>&gt; (apply (case-=
 lambda) (list &#39;x))</div><div>#&lt;case-lambda-procedure&gt;: arity mism=
 atch;</div>
 <div>=C2=A0the expected number of arguments does not match the given number=
 </div><div>=C2=A0 given: 1</div><div>=C2=A0 arguments...:</div><div>=C2=A0 =
 =C2=A0&#39;x</div><div>=C2=A0 context...:</div><div>=C2=A0 =C2=A0/Users/rob=
 by/git/plt/collects/racket/private/misc.rkt:87:7</div>
 <div><br></div><div><br></div><div style>Robby</div></div></div>
 
 --089e0111be00e1a90604d9b627ef--
Responsible changed from "nobody" to "robby" by robby@racket-lang.org at Sat, 06 Apr 2013 20:04:26 -0400
Reason>>> A commit by robby@racket-lang.org was marked as relevant
  http://git.racket-lang.org/plt/commit/4632899b81
  | get a better name for certain contract error messages
  | 
  | related to PR 13665

Responsible changed from "robby" to "endobson" by endobson@racket-lang.org at Thu, 25 Apr 2013 11:44:16 -0400
Reason>>> A commit by endobson@racket-lang.org has resolved this report
  http://git.racket-lang.org/plt/commit/b5b13222c8
State changed from "open" to "closed" by endobson@racket-lang.org at Thu, 25 Apr 2013 11:44:16 -0400
Reason>>> A commit by endobson@racket-lang.org has resolved this report
  http://git.racket-lang.org/plt/commit/b5b13222c8
  | Make type->contract correctly respect variance.
  | 
  | Track whether we are protecting values from the typed side and/or the
  | untyped side.
  | 
  | Closes PR 13662.
  | Closes PR 13663.
  | Closes PR 13665.

